devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy

```diff
@@ -81,41 +81,63 @@ metadata {
 // parse events into attributes
 def parse(String description) {
 	log.debug "Parse description $description"
+	def result = null
 	def map = [:]
-	if (description?.startsWith("read attr -")) {
-		def descMap = parseDescriptionAsMap(description)
-		log.debug "Desc Map: $descMap"
-		if (descMap.cluster == "0201" && descMap.attrId == "0000") {
-			log.debug "TEMP"
-			map.name = "temperature"
-			map.value = getTemperature(descMap.value)
-			map.unit = temperatureScale
-		} else if (descMap.cluster == "0201" && descMap.attrId == "0011") {
-			log.debug "COOLING SETPOINT"
-			map.name = "coolingSetpoint"
-			map.value = getTemperature(descMap.value)
-			map.unit = temperatureScale
-		} else if (descMap.cluster == "0201" && descMap.attrId == "0012") {
-			log.debug "HEATING SETPOINT"
-			map.name = "heatingSetpoint"
-			map.value = getTemperature(descMap.value)
-			map.unit = temperatureScale
-		} else if (descMap.cluster == "0201" && descMap.attrId == "001c") {
-			log.debug "MODE"
-			map.name = "thermostatMode"
-			map.value = getModeMap()[descMap.value]
-		} else if (descMap.cluster == "0202" && descMap.attrId == "0000") {
-			log.debug "FAN MODE"
-			map.name = "thermostatFanMode"
-			map.value = getFanModeMap()[descMap.value]
-		}
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	log.debug "Desc Map: $descMap"
+	if (descMap.cluster == "0201" && descMap.attrId == "0000") {
```

Comment:
I would wrap all this in a check to see if descMap is non-empty.  If the description isn't parseable as an attribute value you could get a null pointer exceptions on these checks.